drivers/spi/spi_shell.c

```diff
@@ -4,54 +4,275 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <string.h>
 #include <errno.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <zephyr/drivers/spi.h>
 #include <zephyr/shell/shell.h>
 #include <zephyr/sys/util.h>
+#include <zephyr/devicetree.h>
 
-#define TXRX_ARGV_BYTES     (1)
-#define CONF_ARGV_DEV       (1)
+#define TXRX_ARGV_SPI_DEV (1)
+#define TXRX_ARGV_BYTES   (2)
+
+#define CONF_ARGV_SPI_DEV   (1)
 #define CONF_ARGV_FREQUENCY (2)
 #define CONF_ARGV_SETTINGS  (3)
 
-#define CS_ARGV_GPIO_DEV    (1)
-#define CS_ARGV_GPIO_PIN    (2)
-#define CS_ARGV_GPIO_FLAGS  (3)
+#define CS_ARGV_SPI_DEV    (1)
+#define CS_ARGV_GPIO_DEV   (2)
+#define CS_ARGV_GPIO_PIN   (3)
+#define CS_ARGV_GPIO_FLAGS (4)
 
 /* Maximum bytes we can write and read at once */
 #define MAX_SPI_BYTES MIN((CONFIG_SHELL_ARGC_MAX - TXRX_ARGV_BYTES), 32)
 
-static struct device *spi_device;
-static struct spi_config config = {.frequency = 1000000,
-				   .operation = SPI_OP_MODE_MASTER | SPI_WORD_SET(8)};
+#define NUM_EMPTY_SPI_THINGS 128
```

Comment:
these probably can be reduced to 32 or less (now it eats like 4K or memory). Also can be made as Kconfg parameter.  Also, I think this and `NUM_EMPTY_MAPS` can be one constant

Reply:
Changed this into a Kconfig, with default value of 32 and joined the constants.